src/Opdex.Platform.WebApi/IndexerBackgroundService.cs

```diff
@@ -10,127 +10,147 @@
+            mediator = scope.ServiceProvider.GetRequiredService<IMediator>();
         }
 
-        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+        using (_logger.BeginScope("Indexer"))
         {
-            var started = false;
-            var unavailable = false;
-
-            IMediator mediator;
-            using var scope = _services.CreateScope();
+            while (!cancellationToken.IsCancellationRequested)
             {
-                mediator = scope.ServiceProvider.GetRequiredService<IMediator>();
-            }
+                if (started)
+                {
+                    // delay 30 seconds when indexing services are unavailable (db flag off)
+                    // delay 8 to 12 seconds when indexing is available
+                    var seconds = unavailable ? 30 : new Random().Next(8, 13);
 
-            using (_logger.BeginScope("Indexer"))
-            {
-                while (!cancellationToken.IsCancellationRequested)
+                    await Task.Delay(TimeSpan.FromSeconds(seconds), cancellationToken);
+                }
+
+                try
                 {
-                    if (started)
+                    var indexLock = await mediator.Send(new RetrieveIndexerLockQuery(), cancellationToken);
+                    if (!indexLock.Available)
                     {
-                        // delay 30 seconds when indexing services are unavailable (db flag off)
-                        // delay 8 to 12 seconds when indexing is available
-                        var seconds = unavailable ? 30 : new Random().Next(8, 13);
-
-                        await Task.Delay(TimeSpan.FromSeconds(seconds), cancellationToken);
+                        _logger.LogWarning("Indexing services unavailable");
+                        unavailable = true;
+                        continue;
                     }
 
-                    try
+                    if (indexLock.Locked)
                     {
-                        var indexLock = await mediator.Send(new RetrieveIndexerLockQuery(), cancellationToken);
-                        if (!indexLock.Available)
-                        {
-                            _logger.LogWarning("Indexing services unavailable");
-                            unavailable = true;
-                            continue;
-                        }
+                        var stuckDuringRewind = indexLock.Reason == IndexLockReason.Rewind && indexLock.ModifiedDate.AddMinutes(1) < DateTime.UtcNow;
+                        var isSameInstanceReprocessing = indexLock.Reason == IndexLockReason.Index && indexLock.InstanceId == _opdexConfiguration.InstanceId;
 
-                        if (indexLock.Locked)
+                        var totalSecondsLocked = DateTime.UtcNow.Subtract(indexLock.ModifiedDate).TotalSeconds;
+
+                        if (!stuckDuringRewind && !isSameInstanceReprocessing)
```

Comment:
What do you think about just logging errors w/ alerts for stuck rewinds over a minute? 

I'm a bit nervous about forcing an unlock around this area and it might be good to see how often it would happen prior to us just letting it automatically go wild.

Reply:
Good idea, I'll make sure we've got an alert set up for this